File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day15ScienceForHungryPeopleTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day15ScienceForHungryPeopleTest  extends FlatSpec with Matchers {
  "2015 Day 15 - Input" should "calculate answers" in {
    val dp = new Day15ScienceForHungryPeople("/twenty_fifteen/Day15-ScienceForHungryPeople-input.txt")

    dp.part1Answer should be(222870)
    dp.part2Answer should be(117936)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day15ScienceForHungryPeople("/twenty_fifteen/Day15-ScienceForHungryPeople-example#1.txt")

    dp.part1Answer should be(62842880)
    dp.part2Answer should be(57600000)
  }
}

File: java-advent-of-code/src/test/java/advent/twenty_fifteen/Day15ScienceForHungryPeopleTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day15ScienceForHungryPeopleTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day15ScienceForHungryPeople("/twenty_fifteen/Day15-ScienceForHungryPeople-input.txt");

        assertEquals(222870, dp.getPart1Answer().intValue());
        assertEquals(117936, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day15ScienceForHungryPeople("/twenty_fifteen/Day15-ScienceForHungryPeople-example#1.txt");

        assertEquals(62842880, dp.getPart1Answer().intValue());
        assertEquals(57600000, dp.getPart2Answer().intValue());
    }
}